lib/src/asset_graph/node.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import '../asset/id.dart';
+import '../builder/builder.dart';
+
+/// A node in the asset graph.
+///
+/// This class specifically represents normal (ie: non-generated) assets.
+class AssetNode {
+  /// The asset this node represents.
+  final AssetId id;
+
+  /// The [AssetId]s of all generated assets which depend on this node.
+  final outputs = new Set<AssetId>();
```

Comment:
This seems fine for now, but I want to be careful about conflating information about the graph structure with details of the build system.

It feels like we should be able to swap out the graph implementation relatively easily without changing much about the rest of the implementation.

Not sure we need to do anything about this now but I wanted to bring it up.